diff --git a/CMakeLists.txt b/CMakeLists.txt
index f573c54..4db288b 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -3,13 +3,9 @@ cmake_minimum_required(VERSION 3.5)
 project(rcutils)
 
 # Default to C11
-if(NOT CMAKE_C_STANDARD)
-  set(CMAKE_C_STANDARD 11)
-endif()
+set(CMAKE_C_STANDARD 11)
 # Default to C++14
-if(NOT CMAKE_CXX_STANDARD)
-  set(CMAKE_CXX_STANDARD 14)
-endif()
+set(CMAKE_CXX_STANDARD 14)
 
 include(CheckLibraryExists)
 
@@ -131,6 +127,10 @@ endif()
 
 target_link_libraries(${PROJECT_NAME} ${CMAKE_DL_LIBS})
 
+if (UNIX AND NOT APPLE)
+  target_link_libraries(${PROJECT_NAME} rt)
+endif()
+
 # Needed if pthread is used for thread local storage.
 if(IOS AND IOS_SDK_VERSION LESS 10.0)
   ament_export_libraries(pthread)
@@ -595,6 +595,9 @@ endif()
 ament_export_dependencies(ament_cmake)
 ament_export_include_directories(include)
 ament_export_libraries(${PROJECT_NAME} ${CMAKE_DL_LIBS})
+if (UNIX AND NOT APPLE)
+  ament_export_libraries(${PROJECT_NAME} rt)
+endif()
 ament_export_targets(${PROJECT_NAME})
 ament_package()
 
