diff --git a/nav2_util/src/dump_params.cpp b/nav2_util/src/dump_params.cpp
index 1908d65ead..49fb5b0438 100644
--- a/nav2_util/src/dump_params.cpp
+++ b/nav2_util/src/dump_params.cpp
@@ -12,7 +12,9 @@
 // See the License for the specific language governing permissions and
 // limitations under the License.
 
+#ifndef _WIN32
 #include <libgen.h>
+#endif
 
 #include <iomanip>
 #include <iostream>
diff --git a/nav2_util/test/test_lifecycle_cli_node.cpp b/nav2_util/test/test_lifecycle_cli_node.cpp
index 2177412e7d..50a458cb1f 100644
--- a/nav2_util/test/test_lifecycle_cli_node.cpp
+++ b/nav2_util/test/test_lifecycle_cli_node.cpp
@@ -23,6 +23,10 @@
 #include "nav2_util/node_thread.hpp"
 #include "rclcpp/rclcpp.hpp"
 
+#ifdef _WIN32
+#include <windows.h>
+#endif
+
 class DummyNode : public nav2_util::LifecycleNode
 {
 public:
@@ -80,7 +84,11 @@ TEST(LifeycleCLI, fails_no_node_name)
   Handle handle;
   auto rc = system("ros2 run nav2_util lifecycle_bringup");
   (void)rc;
+#ifdef _WIN32
+  Sleep(1000);
+#else
   sleep(1);
+#endif
   // check node didn't mode
   EXPECT_EQ(handle.node->activated, false);
   SUCCEED();
@@ -90,7 +98,11 @@ TEST(LifeycleCLI, succeeds_node_name)
 {
   Handle handle;
   auto rc = system("ros2 run nav2_util lifecycle_bringup nav2_test_cli");
+#ifdef _WIN32
+  Sleep(3000);
+#else
   sleep(3);
+#endif
   // check node moved
   (void)rc;
   EXPECT_EQ(handle.node->activated, true);
