diff --git a/src/move_group_sequence_action.cpp b/src/move_group_sequence_action.cpp
index bd106506..cb8b8b53 100644
--- a/src/move_group_sequence_action.cpp
+++ b/src/move_group_sequence_action.cpp
@@ -164,7 +164,7 @@ void MoveGroupSequenceAction::executeSequenceCallbackPlanAndExecute(
   opt.before_execution_callback_ = boost::bind(&MoveGroupSequenceAction::startMoveExecutionCallback, this);
 
   opt.plan_callback_ =
-      boost::bind(&MoveGroupSequenceAction::planUsingSequenceManager, this, boost::cref(goal->request), _1);
+      boost::bind(&MoveGroupSequenceAction::planUsingSequenceManager, this, boost::cref(goal->request), boost::placeholders::_1);
 
   if (goal->planning_options.look_around && context_->plan_with_sensing_)
   {
