diff --git a/cmake/modules/FindLuaGoogle.cmake b/cmake/modules/FindLuaGoogle.cmake
index c76b8cc..4c817f8 100644
--- a/cmake/modules/FindLuaGoogle.cmake
+++ b/cmake/modules/FindLuaGoogle.cmake
@@ -70,7 +70,7 @@ unset(_lua_append_versions)
 
 # this is a function only to have all the variables inside go away automatically
 function(_lua_set_version_vars)
-    set(LUA_VERSIONS5 5.3 5.2)
+    set(LUA_VERSIONS5 5.4 5.3 5.2)
 
     if (Lua_FIND_VERSION_EXACT)
         if (Lua_FIND_VERSION_COUNT GREATER 1)
