diff --git a/bondcpp/CMakeLists.txt b/bondcpp/CMakeLists.txt
index f84c947..f0d1b00 100644
--- a/bondcpp/CMakeLists.txt
+++ b/bondcpp/CMakeLists.txt
@@ -16,10 +16,8 @@ find_package(rclcpp_lifecycle REQUIRED)
 find_package(smclib REQUIRED)
 find_package(Threads REQUIRED)
 
-find_package(PkgConfig REQUIRED)
-pkg_check_modules(UUID REQUIRED uuid)
-# we have to find the absolute path to uuid as target_link_directories is not available before cmake 3.13
-find_library(uuid_ABS_PATH ${UUID_LIBRARIES} PATHS ${UUID_LIBRARY_DIRS})
+find_path(UUID_INCLUDE_DIRS uuid/uuid.h)
+find_library(UUID_LIBRARIES NAMES uuid PATH)
 
 add_library(${PROJECT_NAME}
   SHARED
@@ -34,7 +32,7 @@ ament_target_dependencies(${PROJECT_NAME}
   rclcpp
   rclcpp_lifecycle
   smclib)
-target_link_libraries(${PROJECT_NAME} ${uuid_ABS_PATH})
+target_link_libraries(${PROJECT_NAME} ${UUID_LIBRARIES})
 
 install(
   TARGETS ${PROJECT_NAME}
