diff --git a/CMakeLists.txt b/CMakeLists.txt
index 2dca676d..523e3de7 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -237,6 +237,7 @@ target_link_libraries(rviz_common
   rviz_ogre_vendor::OgreOverlay
   rviz_rendering::rviz_rendering
   Qt5::Widgets
+  rviz_assimp_vendor::rviz_assimp_vendor
 )
 
 ament_target_dependencies(rviz_common
@@ -245,7 +246,6 @@ ament_target_dependencies(rviz_common
   pluginlib
   rclcpp
   resource_retriever
-  rviz_assimp_vendor
   sensor_msgs
   std_msgs
   tf2
