diff --git a/CMakeLists.txt b/CMakeLists.txt
index 9491f51480..c9298c2ca1 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -35,6 +35,9 @@ add_library(${MOVEIT_LIB_NAME} SHARED
   src/interaction_handler.cpp
   src/robot_interaction.cpp
 )
+include(GenerateExportHeader)
+generate_export_header(${MOVEIT_LIB_NAME})
+target_include_directories(${MOVEIT_LIB_NAME} PUBLIC $<BUILD_INTERFACE:${CMAKE_CURRENT_BINARY_DIR}>)
 set_target_properties(${MOVEIT_LIB_NAME} PROPERTIES VERSION "${${PROJECT_NAME}_VERSION}")
 ament_target_dependencies(${MOVEIT_LIB_NAME} ${THIS_PACKAGE_INCLUDE_DEPENDS})
 
@@ -56,6 +59,7 @@ install(
 )
 
 install(DIRECTORY include/ DESTINATION include)
+install(FILES ${CMAKE_CURRENT_BINARY_DIR}/${MOVEIT_LIB_NAME}_export.h DESTINATION include)
 
 install(DIRECTORY resources DESTINATION share/${PROJECT_NAME})
 
diff --git a/include/moveit/robot_interaction/kinematic_options_map.h b/include/moveit/robot_interaction/kinematic_options_map.h
index 12a9c9203d..d00a29541a 100644
--- a/include/moveit/robot_interaction/kinematic_options_map.h
+++ b/include/moveit/robot_interaction/kinematic_options_map.h
@@ -40,11 +40,13 @@
 #include <boost/thread.hpp>
 #include <boost/function.hpp>
 
+#include "moveit_robot_interaction_export.h"
+
 namespace robot_interaction
 {
 // Maintains a set of KinematicOptions with a key/value mapping and a default
 // value.
-class KinematicOptionsMap
+class MOVEIT_ROBOT_INTERACTION_EXPORT KinematicOptionsMap
 {
 public:
   /// Constructor - set all options to reasonable default values.
