diff --git a/CMakeLists.txt b/CMakeLists.txt
index 348085e..8786cce 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -26,7 +26,6 @@ ament_package_xml()
 include_directories(include ${rclcpp_INCLUDE_DIRS}
   ${rmw_implementation_INCLUDE_DIRS}
   ${Boost_INCLUDE_DIR}
-  ${TinyXML2_INCLUDE_DIRS}
   ${urdfdom_headers_INCLUDE_DIRS}
   ${urdf_INCLUDE_DIRS}
   ${std_msgs_INCLUDE_DIRS}
@@ -39,19 +38,19 @@ add_library(${PROJECT_NAME} SHARED
 set_target_properties(${PROJECT_NAME} PROPERTIES
   VERSION ${${PROJECT_NAME}_VERSION})
 
-# System dependencies
-target_link_libraries(${PROJECT_NAME} PUBLIC ${TinyXML2_LIBRARIES})
-
 # Ament dependencies
 ament_target_dependencies(${PROJECT_NAME} PUBLIC
+  TinyXML2
   console_bridge
-  urdfdom_headers
   urdf
+  urdfdom_headers
 )
 
 install(TARGETS ${PROJECT_NAME}
   EXPORT ${PROJECT_NAME}
-  DESTINATION lib
+  ARCHIVE DESTINATION lib
+  LIBRARY DESTINATION lib
+  RUNTIME DESTINATION bin
 )
 install(DIRECTORY include/${PROJECT_NAME}/
   DESTINATION include/${PROJECT_NAME}
@@ -64,6 +63,7 @@ install(PROGRAMS
 )
 
 set_target_properties(${PROJECT_NAME} PROPERTIES WINDOWS_EXPORT_ALL_SYMBOLS TRUE)
+target_compile_definitions(${PROJECT_NAME} PRIVATE "SRDFDOM_BUILDING_DLL")
 
 if(BUILD_TESTING)
   find_package(ament_cmake_gtest REQUIRED)
@@ -86,8 +86,10 @@ if(BUILD_TESTING)
 endif()
 
 ament_export_include_directories(include)
-ament_export_libraries(${PROJECT_NAME} ${TinyXML2_LIBRARIES})
+ament_export_libraries(${PROJECT_NAME})
 ament_export_dependencies(console_bridge)
 ament_export_dependencies(urdfdom_headers)
 ament_export_dependencies(urdf)
+ament_export_dependencies(tinyxml2_vendor)
+ament_export_dependencies(TinyXML2)
 ament_package()
diff --git a/include/srdfdom/model.h b/include/srdfdom/model.h
index 7c1a69d..985f07e 100644
--- a/include/srdfdom/model.h
+++ b/include/srdfdom/model.h
@@ -45,11 +45,13 @@
 #include <memory>
 #include <tinyxml2.h>
 
+#include "visibility_control.h"
+
 /// Main namespace
 namespace srdf
 {
 /** \brief Representation of semantic information about the robot */
-class Model
+class SRDFDOM_PUBLIC Model
 {
 public:
   Model()
diff --git a/include/srdfdom/visibility_control.h b/include/srdfdom/visibility_control.h
new file mode 100644
index 0000000..2c46c73
--- /dev/null
+++ b/include/srdfdom/visibility_control.h
@@ -0,0 +1,35 @@
+#ifndef SRDFDOM__VISIBILITY_CONTROL_H_
+#define SRDFDOM__VISIBILITY_CONTROL_H_
+
+// This logic was borrowed (then namespaced) from the examples on the gcc wiki:
+//     https://gcc.gnu.org/wiki/Visibility
+
+#if defined _WIN32 || defined __CYGWIN__
+#ifdef __GNUC__
+#define SRDFDOM_EXPORT __attribute__((dllexport))
+#define SRDFDOM_IMPORT __attribute__((dllimport))
+#else
+#define SRDFDOM_EXPORT __declspec(dllexport)
+#define SRDFDOM_IMPORT __declspec(dllimport)
+#endif
+#ifdef SRDFDOM_BUILDING_DLL
+#define SRDFDOM_PUBLIC SRDFDOM_EXPORT
+#else
+#define SRDFDOM_PUBLIC SRDFDOM_IMPORT
+#endif
+#define SRDFDOM_PUBLIC_TYPE SRDFDOM_PUBLIC
+#define SRDFDOM_LOCAL
+#else
+#define SRDFDOM_EXPORT __attribute__((visibility("default")))
+#define SRDFDOM_IMPORT
+#if __GNUC__ >= 4
+#define SRDFDOM_PUBLIC __attribute__((visibility("default")))
+#define SRDFDOM_LOCAL __attribute__((visibility("hidden")))
+#else
+#define SRDFDOM_PUBLIC
+#define SRDFDOM_LOCAL
+#endif
+#define SRDFDOM_PUBLIC_TYPE
+#endif
+
+#endif  // SRDFDOM__VISIBILITY_CONTROL_H_
