diff --git a/CMakeLists.txt b/CMakeLists.txt
index 99c9e2d..67d3abc 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -14,7 +14,18 @@ else()
   set(PYTHON_INSTALL_DIR "lib/python${PYTHON_MAJOR_MINOR}/site-packages")
   set(SHELL_EXT "sh")
 endif()
-set(AMENT_PACKAGE_DIR "${CMAKE_INSTALL_PREFIX}/${PYTHON_INSTALL_DIR}/ament_package")
+
+if ($ENV{CONDA_BUILD_CROSS_COMPILATION} EQ "1")
+  execute_process(
+    COMMAND python3 -c "import sys; print(sys.prefix)"
+    OUTPUT_VARIABLE _PYTHON_SYSPREFIX
+    OUTPUT_STRIP_TRAILING_WHITESPACE
+  )
+  set(AMENT_PACKAGE_DIR "${_PYTHON_SYSPREFIX}/ament_package")
+else()
+  set(AMENT_PACKAGE_DIR "${CMAKE_INSTALL_PREFIX}/${PYTHON_INSTALL_DIR}/ament_package")
+endif()
+
 if(NOT EXISTS "${AMENT_PACKAGE_DIR}")
   # Check for an .egg-link file and use the listed directory if it exists
   get_filename_component(AMENT_PACKAGE_EGG_LINK "${AMENT_PACKAGE_DIR}" DIRECTORY)
