diff --git a/nav2_waypoint_follower/CMakeLists.txt b/nav2_waypoint_follower/CMakeLists.txt
index a3b46942b..b0930e1aa 100644
--- a/nav2_waypoint_follower/CMakeLists.txt
+++ b/nav2_waypoint_follower/CMakeLists.txt
@@ -25,7 +25,11 @@ find_package(pluginlib REQUIRED)
 
 nav2_package()
 
-link_libraries(stdc++fs)
+if(NOT APPLE)
+  link_libraries(stdc++fs)
+else()
+
+endif()
 
 include_directories(
   include
