diff --git a/include/moveit_servo/servo_calcs.h b/include/moveit_servo/servo_calcs.h
index 097069a153..c2c6212a5d 100644
--- a/include/moveit_servo/servo_calcs.h
+++ b/include/moveit_servo/servo_calcs.h
@@ -333,7 +333,7 @@ class ServoCalcs
 
   const int gazebo_redundant_message_count_ = 30;
 
-  uint num_joints_;
+  unsigned int num_joints_;
 
   // True -> allow drift in this dimension. In the command frame. [x, y, z, roll, pitch, yaw]
   std::array<bool, 6> drift_dimensions_ = { { false, false, false, false, false, false } };

