diff --git a/planning_scene_rviz_plugin/CMakeLists.txt b/planning_scene_rviz_plugin/CMakeLists.txt
index 5b65ecdba6..b987a90b9e 100644
--- a/planning_scene_rviz_plugin/CMakeLists.txt
+++ b/planning_scene_rviz_plugin/CMakeLists.txt
@@ -3,6 +3,9 @@ set(MOVEIT_LIB_NAME moveit_planning_scene_rviz_plugin)
 add_library(${MOVEIT_LIB_NAME}_core SHARED
   src/planning_scene_display.cpp
   include/moveit/planning_scene_rviz_plugin/planning_scene_display.h)
+include(GenerateExportHeader)
+generate_export_header(${MOVEIT_LIB_NAME}_core)
+target_include_directories(${MOVEIT_LIB_NAME}_core PUBLIC $<BUILD_INTERFACE:${CMAKE_CURRENT_BINARY_DIR}>)
 set_target_properties(${MOVEIT_LIB_NAME}_core PROPERTIES VERSION "${${PROJECT_NAME}_VERSION}")
 target_link_libraries(${MOVEIT_LIB_NAME}_core moveit_rviz_plugin_render_tools)
 ament_target_dependencies(${MOVEIT_LIB_NAME}_core
@@ -27,3 +30,4 @@ ament_target_dependencies(${MOVEIT_LIB_NAME}
 target_include_directories(${MOVEIT_LIB_NAME} PRIVATE "${OGRE_PREFIX_DIR}/include")
 
 install(DIRECTORY include/ DESTINATION include)
+install(FILES ${CMAKE_CURRENT_BINARY_DIR}/${MOVEIT_LIB_NAME}_core_export.h DESTINATION include)
diff --git a/planning_scene_rviz_plugin/include/moveit/planning_scene_rviz_plugin/planning_scene_display.h b/planning_scene_rviz_plugin/include/moveit/planning_scene_rviz_plugin/planning_scene_display.h
index 5e3ae0d488..7f0fd21c06 100644
--- a/planning_scene_rviz_plugin/include/moveit/planning_scene_rviz_plugin/planning_scene_display.h
+++ b/planning_scene_rviz_plugin/include/moveit/planning_scene_rviz_plugin/planning_scene_display.h
@@ -47,6 +47,8 @@
 #include <rclcpp/rclcpp.hpp>
 #endif
 
+#include "moveit_planning_scene_rviz_plugin_core_export.h"
+
 namespace Ogre
 {
 class SceneNode;
@@ -66,7 +68,7 @@ class EnumProperty;
 
 namespace moveit_rviz_plugin
 {
-class PlanningSceneDisplay : public rviz_common::Display
+class MOVEIT_PLANNING_SCENE_RVIZ_PLUGIN_CORE_EXPORT PlanningSceneDisplay : public rviz_common::Display
 {
   Q_OBJECT
 