diff --git a/CMakeLists.txt b/CMakeLists.txt
index ceb8b528..7eceed9e 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -56,7 +49,7 @@ find_package(pluginlib REQUIRED)
 set(VERSION_FILE_PATH ${CMAKE_BINARY_DIR}/include)
 
 set(THIS_PACKAGE_INCLUDE_DIRS
-    background_processing/include
+    # background_processing/include
     exceptions/include
     backtrace/include
     collision_detection/include
@@ -88,7 +81,7 @@ set(THIS_PACKAGE_INCLUDE_DIRS
 )
 
 set(THIS_PACKAGE_LIBRARIES
-    moveit_background_processing
+    # moveit_background_processing
     moveit_butterworth_filter
     moveit_collision_distance_field
     moveit_collision_detection
@@ -158,7 +151,7 @@ message(STATUS " *** Building MoveIt ${MOVEIT_VERSION} ***")
 configure_file("version/version.h.in" "${VERSION_FILE_PATH}/moveit/version.h")
 install(FILES "${VERSION_FILE_PATH}/moveit/version.h" DESTINATION include/moveit)
 
-add_subdirectory(background_processing)
+# add_subdirectory(background_processing)
 add_subdirectory(backtrace)
 add_subdirectory(collision_distance_field)
 add_subdirectory(constraint_samplers)
