diff --git a/nav2_rviz_plugins/include/nav2_rviz_plugins/nav2_panel.hpp b/nav2_rviz_plugins/include/nav2_rviz_plugins/nav2_panel.hpp
index 58b53fdf8c..a656cf9220 100644
--- a/nav2_rviz_plugins/include/nav2_rviz_plugins/nav2_panel.hpp
+++ b/nav2_rviz_plugins/include/nav2_rviz_plugins/nav2_panel.hpp
@@ -17,6 +17,7 @@
 
 #include <QtWidgets>
 #include <QBasicTimer>
+#undef NO_ERROR
 
 #include <memory>
 #include <string>
