diff --git a/perception/CMakeLists.txt b/perception/CMakeLists.txt
index 27ce4ca893..f2fef34935 100644
--- a/perception/CMakeLists.txt
+++ b/perception/CMakeLists.txt
@@ -28,7 +28,11 @@ if(WITH_OPENGL)
     set(SYSTEM_GL_LIBRARIES ${GLEW_LIBRARIES} GLEW::GLEW FreeGLUT::freeglut)
   else()
     find_package(GLUT REQUIRED)
-    set(SYSTEM_GL_LIBRARIES ${GLEW_LIBRARIES} GLUT::GLUT)
+    if(WIN32)
+      set(SYSTEM_GL_LIBRARIES GLEW::glew GLUT::GLUT)
+    else()
+      set(SYSTEM_GL_LIBRARIES ${GLEW_LIBRARIES} GLUT::GLUT)
+    endif()
   endif()
   set(perception_GL_INCLUDE_DIRS "mesh_filter/include" "depth_image_octomap_updater/include")
   set(SYSTEM_GL_INCLUDE_DIRS ${GLEW_INCLUDE_DIR} ${GLUT_INCLUDE_DIR})
diff --git a/perception/depth_image_octomap_updater/src/depth_image_octomap_updater.cpp b/perception/depth_image_octomap_updater/src/depth_image_octomap_updater.cpp
index 4edc9a3758..dc186f44e4 100644
--- a/perception/depth_image_octomap_updater/src/depth_image_octomap_updater.cpp
+++ b/perception/depth_image_octomap_updater/src/depth_image_octomap_updater.cpp
@@ -44,6 +44,7 @@
 #include <stdint.h>
 
 #include <memory>
+#include <boost/bind.hpp>
 
 namespace occupancy_map_monitor
 {
diff --git a/perception/mesh_filter/CMakeLists.txt b/perception/mesh_filter/CMakeLists.txt
index 6a83af3974..54b973d317 100644
--- a/perception/mesh_filter/CMakeLists.txt
+++ b/perception/mesh_filter/CMakeLists.txt
@@ -7,6 +7,9 @@ add_library(${MOVEIT_LIB_NAME} SHARED
   src/gl_renderer.cpp
   src/gl_mesh.cpp
 )
+include(GenerateExportHeader)
+generate_export_header(${MOVEIT_LIB_NAME})
+target_include_directories(${MOVEIT_LIB_NAME} PUBLIC $<BUILD_INTERFACE:${CMAKE_CURRENT_BINARY_DIR}>)
 set_target_properties(${MOVEIT_LIB_NAME} PROPERTIES VERSION "${${PROJECT_NAME}_VERSION}")
 ament_target_dependencies(${MOVEIT_LIB_NAME}
   rclcpp
@@ -40,3 +43,4 @@ target_link_libraries(${MOVEIT_LIB_NAME} ${gl_LIBS} ${SYSTEM_GL_LIBRARIES})
 # endif()
 
 install(DIRECTORY include/ DESTINATION include)
+install(FILES ${CMAKE_CURRENT_BINARY_DIR}/${MOVEIT_LIB_NAME}_export.h DESTINATION include)
diff --git a/perception/mesh_filter/include/moveit/mesh_filter/gl_renderer.h b/perception/mesh_filter/include/moveit/mesh_filter/gl_renderer.h
index 6f4f9243fd..92e3441c4c 100644
--- a/perception/mesh_filter/include/moveit/mesh_filter/gl_renderer.h
+++ b/perception/mesh_filter/include/moveit/mesh_filter/gl_renderer.h
@@ -48,6 +48,9 @@
 #include <mutex>
 #include <map>
 
+#undef near
+#undef far
+
 namespace mesh_filter
 {
 MOVEIT_CLASS_FORWARD(GLRenderer);  // Defines GLRendererPtr, ConstPtr, WeakPtr... etc
diff --git a/perception/mesh_filter/include/moveit/mesh_filter/stereo_camera_model.h b/perception/mesh_filter/include/moveit/mesh_filter/stereo_camera_model.h
index 5ffd8174d3..cd4a0a7732 100644
--- a/perception/mesh_filter/include/moveit/mesh_filter/stereo_camera_model.h
+++ b/perception/mesh_filter/include/moveit/mesh_filter/stereo_camera_model.h
@@ -39,13 +39,15 @@
 #include <moveit/mesh_filter/sensor_model.h>
 #include <string>
 
+#include "moveit_mesh_filter_export.h"
+
 namespace mesh_filter
 {
 /**
  * \brief Model for Disparity based devices. E.g stereo camera systems or OpenNI compatible devices
  * \author Suat Gedikli <gedikli@willowgarage.com>
  */
-class StereoCameraModel : public SensorModel
+class MOVEIT_MESH_FILTER_EXPORT StereoCameraModel : public SensorModel
 {
 public:
   /**
diff --git a/perception/pointcloud_octomap_updater/src/pointcloud_octomap_updater.cpp b/perception/pointcloud_octomap_updater/src/pointcloud_octomap_updater.cpp
index 4007b4da4f..cd163cb601 100644
--- a/perception/pointcloud_octomap_updater/src/pointcloud_octomap_updater.cpp
+++ b/perception/pointcloud_octomap_updater/src/pointcloud_octomap_updater.cpp
@@ -45,6 +45,7 @@
 #include <tf2_ros/create_timer_ros.h>
 
 #include <memory>
+#include <boost/bind.hpp>
 
 namespace occupancy_map_monitor
 {

