diff --git a/CMakeLists.txt b/CMakeLists.txt
index 642d32f..0fd9169 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -12,29 +12,11 @@ if(CMAKE_COMPILER_IS_GNUCXX OR CMAKE_CXX_COMPILER_ID MATCHES "Clang")
   add_compile_options(-Wall -Wextra)
 endif()
 
-if(NOT ANDROID)
-  find_package(PythonLibs)
-  if(PYTHONLIBS_VERSION_STRING VERSION_LESS 3)
-    find_package(Boost REQUIRED python)
-  else()
-    find_package(Boost QUIET)
-    if(Boost_VERSION LESS 106500)
-      find_package(Boost REQUIRED python)
-    else()
-      # This is a bit of a hack to suppress a warning
-      #   No header defined for python3; skipping header check
-      # Which should only affect Boost versions < 1.67
-      # Resolution for newer versions:
-      #  https://gitlab.kitware.com/cmake/cmake/issues/16391
-      if (Boost_VERSION LESS 106700)
-        set(_Boost_PYTHON3_HEADERS "boost/python.hpp")
-      endif()
-      find_package(Boost COMPONENTS python3 REQUIRED)
-    endif()
-  endif()
-else()
-  find_package(Boost REQUIRED)
-endif()
+find_package(PythonLibs REQUIRED)
+string(REPLACE "." ";" VERSION_LIST ${PYTHONLIBS_VERSION_STRING})
+list(GET VERSION_LIST 0 PYTHONLIBS_VERSION_MAJOR)
+list(GET VERSION_LIST 1 PYTHONLIBS_VERSION_MINOR)
+find_package(Boost COMPONENTS python${PYTHONLIBS_VERSION_MAJOR}${PYTHONLIBS_VERSION_MINOR} REQUIRED)
 
 find_package(sensor_msgs REQUIRED)
 
diff --git a/src/CMakeLists.txt b/src/CMakeLists.txt
index 9e2cf45..a9a4194 100644
--- a/src/CMakeLists.txt
+++ b/src/CMakeLists.txt
@@ -10,9 +10,14 @@ ament_target_dependencies(${PROJECT_NAME}
   "OpenCV"
   "sensor_msgs"
 )
+link_directories(${Boost_LIBRARY_DIRS})
 target_link_libraries(${PROJECT_NAME} ${Boost_LIBRARIES} ${OpenCV_LIBRARIES})
 
-install(TARGETS ${PROJECT_NAME} DESTINATION lib)
+install(TARGETS ${PROJECT_NAME}
+  ARCHIVE DESTINATION lib
+  LIBRARY DESTINATION lib
+  RUNTIME DESTINATION bin
+)
 
 install(FILES
   ${CMAKE_CURRENT_BINARY_DIR}/${PROJECT_NAME}/${PROJECT_NAME}_export.h
@@ -56,6 +61,7 @@ endif()
 target_link_libraries(${PROJECT_NAME}_boost
   ${PYTHON_LIBRARIES}
   ${PROJECT_NAME}
+  Boost::python${PYTHONLIBS_VERSION_MAJOR}${PYTHONLIBS_VERSION_MINOR}
 )
 
 set_target_properties(${PROJECT_NAME}_boost PROPERTIES
